test/hip-583/HIP583.js

```diff
@@ -545,4 +545,154 @@ describe("Contract Transfer TX", function () {
+    signers = await ethers.getSigners();
+    tokenCreateContract = await utils.deployTokenCreateContract();
+    tokenQueryContract = await utils.deployTokenQueryContract();
+    tokenTransferContract = await utils.deployTokenTransferContract();
+    erc20Contract = await utils.deployERC20Contract();
+    erc721Contract = await utils.deployERC721Contract();
+  });
+
+  const bootstrapHollowAccount = async function (signer, hollowWallet, tokenCreateContract, tokenAddress) {
+    await signer.sendTransaction({
+      to: hollowWallet.address,
+      value: ethers.utils.parseEther("100"),
+      gasLimit: 1_000_000,
+    });
+
+    const hollowWalletTokenCreateContract = await tokenCreateContract.connect(hollowWallet);
+    await (await hollowWalletTokenCreateContract.associateTokenPublic(hollowWallet.address, tokenAddress, {
+      gasLimit: 1_000_000,
+    })).wait();
+    await (await tokenCreateContract.grantTokenKycPublic(tokenAddress, hollowWallet.address)).wait();
+  }
+
+  describe("Fungible Token Test", function () {
+    const amount = 27;
+    let tokenAddress;
+    let hollowWallet;
+
+    before(async function () {
+      tokenAddress = await utils.createFungibleTokenWithSECP256K1AdminKey(tokenCreateContract, signers[0].address, utils.getSignerCompressedPublicKey());
+      await utils.associateToken(tokenCreateContract, tokenAddress, 'TokenCreateContract');
+      await utils.grantTokenKyc(tokenCreateContract, tokenAddress);
+
+      hollowWallet = ethers.Wallet.createRandom().connect(ethers.provider);
+      await bootstrapHollowAccount(signers[0], hollowWallet, tokenCreateContract, tokenAddress)
+    });
+
+    it("should test that hollow account is created and the amount of fungible tokens is correctly transferred via precompile", async function () {
+      const hollowBalanceBefore = await erc20Contract.balanceOf(tokenAddress, hollowWallet.address);
+      await tokenTransferContract.transferTokensPublic(tokenAddress, [signers[0].address, hollowWallet.address], [-amount, amount], {gasLimit: 1_000_000});
+      const hollowBalanceAfter = await erc20Contract.balanceOf(tokenAddress, hollowWallet.address);
+
+      expect(hollowBalanceBefore).to.eq(0);
+      expect(hollowBalanceAfter).to.eq(amount);
+    });
+
+    it("should test that second transfer fungible tokens via precompile to the hollow account is successful", async function () {
+      const hollowBalanceBefore = await erc20Contract.balanceOf(tokenAddress, hollowWallet.address);
+      await tokenTransferContract.transferTokensPublic(tokenAddress, [signers[0].address, hollowWallet.address], [-amount, amount], {gasLimit: 1_000_000});
+      const hollowBalanceAfter = await erc20Contract.balanceOf(tokenAddress, hollowWallet.address);
+
+      expect(hollowBalanceAfter).to.eq(hollowBalanceBefore.add(amount));
+    });
+
+    it("should test that second transfer fungible tokens via precompile to the hollow account is successful", async function () {
```

Comment:
Should this test be here?  The description is the same as the one above but the test body is the same as the one below?